kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/clues.kt

```diff
@@ -28,6 +28,8 @@ inline fun <R> withClue(crossinline clue: () -> Any?, thunk: () -> R): R {
       // this is a special control exception used by coroutines
    } catch (t: TimeoutCancellationException) {
       throw Exceptions.createAssertionError(clueContextAsString() + (t.message ?: ""), t)
+   } catch (e: Exception) {
+      throw Exception(clueContextAsString() + (e.message ?: ""), e)
```

Comment:
Let's throw a very specific Exception here, such as `ExceptionWithClue`, not a generic one. And if we catch it, let's just rethrow it:
```kotlin
catch(c: ExceptionWithClue) { throw c }
```